module/Finna/src/Finna/Search/Factory/L1BackendFactory.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/**
+ * Factory for a L1 backend
+ *
+ * PHP version 7
+ *
+ * Copyright (C) The National Library of Finland 2020.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Search_Factory
+ * @author   Aleksi Peebles <aleksi.peebles@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Site
+ */
+namespace Finna\Search\Factory;
+
+use FinnaSearch\Backend\L1\Connector;
+
+/**
+ * Factory for a L1 backend
+ *
+ * @category VuFind
+ * @package  Search_Factory
+ * @author   Aleksi Peebles <aleksi.peebles@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Site
+ */
+class L1BackendFactory extends SolrDefaultBackendFactory
+{
+    /**
+     * Callback for creating a record driver.
+     *
+     * @var string
+     */
+    protected $createRecordMethod = 'getL1Record';
```

Comment:
Toteutus puuttuu?

Reply:
Tuota näemmä käytetään ainoastaan `VuFind\Search\Factory\SolrDefaultBackendFactory::createBackend()`:issa, mutta `Finna\Search\Factory\SolrDefaultBackendFactory`, josta `L1BackendFactory` periytyy, yliajaa tuon metodin ilman `parent::`-kutsua. Eli onko koko muuttuja itse asiassa tarpeeton, vai lisätäänkö toteutus joka tapauksessa?

Reply:
Hankala kysymys. Tavallaan siis turha muuttuja, mutta ei se oletusarvokaan ihan oikea ole. Anna olla vain noin siltä varalta, että jokin toinen metodi käyttäisi sitä joskus tulevaisuudessa.

Reply:
Varmistan vielä että ymmärsin oikein, eli annan muuttujan olla mutta en lisää toteutusta?

Reply:
Sori, ymmärsin väärin. Meidän toteutuksen pitäisi käyttää myös tuota `$this->createRecordMethod` eikä kovakoodata sitä. Eli pitäisi muuttaa Finnan createBackend.